src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.cs

```diff
@@ -68,16 +173,58 @@ string IDkmClrFormatter.GetUnderlyingString(DkmClrValue value, DkmInspectionCont
         internal abstract bool IsWhitespace(char c);
 
         // Note: We could be less conservative (e.g. "new C()").
-        internal bool NeedsParentheses(string expr)
+        private bool NeedsParentheses(string expr)
         {
-            foreach (var ch in expr)
+            int parens = 0;
+            for (int i = 0; i < expr.Length; i++)
             {
-                if (!this.IsIdentifierPartCharacter(ch) && !this.IsWhitespace(ch) && ch != '.') return true;
+                var ch = expr[i];
+                switch (ch)
+                {
+                    case '(':
+                        // Cast, "(A)b", requires parentheses.
+                        if ((parens == 0) && FollowsCloseParen(expr, i))
+                        {
+                            return true;
+                        }
+                        parens++;
+                        break;
+                    case '[':
```

Comment:
Square brackets are tracked to avoid returning true in all cases where ```NeedsParentheses``` would return true for the expression within brackets (e.g.: ```a[1 + 2]```).